qa/rpc-tests/test_framework/mininode.py

```diff
@@ -660,14 +753,18 @@ def is_valid(self):
 
     def __repr__(self):
         r = ("CTransaction(fOverwintered=%r nVersion=%i nVersionGroupId=0x%08x "
-             "vin=%s vout=%s nLockTime=%i nExpiryHeight=%i"
+             "vin=%s vout=%s nLockTime=%i nExpiryHeight=%i valueBalance=%i "
+             "shieldedSpends=%s shieldedOutputs=%s"
              % (self.fOverwintered, self.nVersion, self.nVersionGroupId,
-                repr(self.vin), repr(self.vout), self.nLockTime, self.nExpiryHeight))
+                repr(self.vin), repr(self.vout), self.nLockTime, self.nExpiryHeight,
+                self.valueBalance, repr(self.shieldedSpends), repr(self.shieldedOutputs)))
         if self.nVersion >= 2:
             r += " vJoinSplit=%s" % repr(self.vJoinSplit)
             if len(self.vJoinSplit) > 0:
                 r += " joinSplitPubKey=%064x joinSplitSig=%064x" \
                     (self.joinSplitPubKey, self.joinSplitSig)
+        if self.nVersion == 4 and not (len(self.shieldedSpends) == 0 and len(self.shieldedOutputs) == 0):
```

Comment:
The `self.nVersion == 4` condition is unnecessary, because `self.shieldedSpends` and `self.shieldedOutputs` will only be set in a version for which they can be parsed.
```suggestion
        if len(self.shieldedSpends) > 0 or len(self.shieldedOutputs) > 0:
```